### PR TITLE
Streamlit + `SUBSCRIBE` demo

### PR DESCRIPTION
**Pull request by bobbyiliev**
As discussed internally, adding a Streamlit + `SUBSCRIBE` demo:

streamlit-subscribe-upsert

### CODE REVIEW

File: streamlit-subscribe/README.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        WITH (
+            HEADERS,
+            SECRET basic_auth_secret
+        )
+        "headers" -> 'authorization' = "basic_auth_secret"
+    );
+```
+
+The public URL for this webhook source is `https://<HOST>/api/webhook/<database>/<schema>/<src_name>`:
+- `<HOST>` is the hostname of the Materialize cluster.
```
Comment (dehume) [inline streamlit-subscribe/README.md:69]: Inconsistent that this capitalized but the others aren't

File: streamlit-subscribe/subscribe.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            # print(f"Row from database: {row}")  # Log the fetched row for debugging
+            updates_queue.put(row)
+
+# Create a background thread to fetch data
+thread = threading.Thread(target=fetch_data)
+thread.start()
+
+# Initialize the session state
+if 'data' not in st.session_state:
+    st.session_state.data = {
```
Comment (dehume) [inline streamlit-subscribe/subscribe.py:25]: Could this just be a `defaultdict(list)` (https://docs.python.org/3/library/collections.html#defaultdict-examples) so you don't have to initialize all the keys?

File: streamlit-subscribe/README.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+Once you have your Materialize webhook source ready, create a `.env` file with the following variables:
+
+```
+# Webhook
+export WEBHOOK_URL=https://<HOST>/api/webhook/<database>/<schema>/<src_name>
+export WEBHOOK_SECRET=some-secret-value
+```
+
+You can use the [sensors.json](sensors.json) schema:
```
Comment (joacoc) [inline streamlit-subscribe/README.md:94]: The json is missing

File: streamlit-subscribe/README.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+3.  Install the required packages:
+
+    ```bash
+    pip install streamlit psycopg altair pandas
+    ```
+
+4.  Set the `DATABASE_URL` environment variable to the connection string of your Materialize instance:
+
+    ```python
+    export DATABASE_URL="postgres://USERNAME:APP_PASSWORD@ID.materialize.cloud:6875/materialize?sslmode=require"
```
Comment (joacoc) [inline streamlit-subscribe/README.md:40]:
```suggestion
    export DATABASE_URL="postgres://<USERNAME>:<APP_PASSWORD>@<HOST>/materialize?sslmode=require"
```

File: streamlit-subscribe/README.md (hunk 3 above)
Comment (bobbyiliev) [reply streamlit-subscribe/README.md:94]: ah yes good point! just pushed it

File: streamlit-subscribe/subscribe.py (hunk 2 above)
Comment (bobbyiliev) [reply streamlit-subscribe/subscribe.py]: Neat! Thank you for suggesting that!

File: streamlit-subscribe/README.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+    ```bash
+    python -m venv venv
+    source .venv/bin/activate
+    # On Windows use `venv\Scripts\activate`
+    ```
+
+3.  Install the required packages:
+
+    ```bash
+    pip install streamlit psycopg altair pandas
```
[the remark this answers is not in the document]
Comment (bobbyiliev) [reply streamlit-subscribe/README.md]: Good points! Just updated this

**Review by dehume — APPROVED** [review]

**Pull request merged**